pkg/utils/protoutils/marshal.go

```diff
@@ -35,6 +35,33 @@ func MarshalBytes(res resources.Resource) ([]byte, error) {
 	return json.Marshal(res)
 }
 
+func UnmarshalYAML(data []byte, into resources.Resource) error {
+	jsn, err := yaml.YAMLToJSON(data)
+	if err != nil {
+		return err
+	}
+
+	if protoInto, ok := into.(proto.Message); ok {
+		return jsonpb.Unmarshal(bytes.NewBuffer(jsn), protoInto)
+	}
+	return json.Unmarshal(data, into)
+}
+
+func MarshalYAML(res resources.Resource) ([]byte, error) {
+	jsn, err := func() ([]byte, error) {
```

Comment:
Any particular reason this is in a closure, not a problem, just seems unecessary and makes it a little harder to read

Reply:
it's not much more readable the other way, but i can change it if u prefer:
```go 

func MarshalYAML(res resources.Resource) ([]byte, error) {
	var jsn []byte
	if pb, ok := res.(proto.Message); ok {
		buf := &bytes.Buffer{}
		if err := jsonpbMarshaler.Marshal(buf, pb); err != nil {
			return nil, err
		}
		jsn = buf.Bytes()
	} else {
		var err error 
		jsn, err = json.Marshal(res)
		if err != nil {
			return nil, err
		}
	}
	return yaml.JSONToYAML(jsn)
}
```

Reply:
Thanks, I appreciate it